oops/backplane/exercise_backplanes.py

```diff
@@ -144,872 +33,220 @@ def show_info(title, array):
-
-    test = bp.where_between(('incidence_angle', 'saturn'), HALFPI,3.2)
-    show_info('Mask of Saturn anti-sunward via where_between()', test)
-
-    test = bp.where_in_front('saturn', 'saturn_main_rings')
-    show_info('Mask of Saturn in front of rings', test)
-
-    test = bp.where_in_back('saturn', 'saturn_main_rings')
-    show_info('Mask of Saturn behind rings', test)
-
-    test = bp.where_inside_shadow('saturn', 'saturn_main_rings')
-    show_info('Mask of Saturn in shadow of rings', test)
-
-    test = bp.where_outside_shadow('saturn', 'saturn_main_rings')
-    show_info('Mask of Saturn outside shadow of rings', test)
-
-    test = bp.where_intercepted('saturn_main_rings')
-    show_info('Mask of rings intercepted', test)
-
-    test = bp.where_in_front('saturn_main_rings', 'saturn')
-    show_info('Mask of rings in front of Saturn', test)
-
-    test = bp.where_in_back('saturn_main_rings', 'saturn')
-    show_info('Mask of rings behind Saturn', test)
-
-    test = bp.where_inside_shadow('saturn_main_rings', 'saturn')
-    show_info('Mask of rings in shadow of Saturn', test)
-
-    test = bp.where_outside_shadow('saturn_main_rings', 'saturn')
-    show_info('Mask of rings outside shadow of Saturn', test)
-
-    test = bp.where_sunward('saturn_main_rings')
-    show_info('Mask of rings sunward', test)
-
-    test = bp.where_antisunward('saturn_main_rings')
-    show_info('Mask of rings anti-sunward', test)
-
-    ########################
-
-    if printing: print('\n********* borders')
-
-    mask = bp.where_intercepted('saturn')
-    test = bp.border_inside(mask)
-    show_info('Border of Saturn intercepted mask, inside', test)
-
-    test = bp.border_outside(mask)
-    show_info('Border of Saturn intercepted mask, outside', test)
-
-    test = bp.border_below(('ring_radius', 'saturn:ring'), 100.e3)
-    show_info('Border of ring radius below 100 kkm', test)
-
-    test = bp.border_atop(('ring_radius', 'saturn:ring'), 100.e3)
-    show_info('Border of ring radius atop 100 kkm', test)
-
-    test = bp.border_above(('ring_radius', 'saturn:ring'), 100.e3)
-    show_info('Border of ring radius above 100 kkm', test)
-
-    test = bp.evaluate(('border_above', ('ring_radius', 'saturn:ring'), 100.e3))
-    show_info('Border of ring radius above 100 kkm via evaluate()', test)
-
-    ########################
-
-    if printing: print('\n********* EMPTY EVENTS')
+    config.LOGGING.off()
 
-    test = bp.where_below(('ring_radius', 'saturn_main_rings'), 10.e3)
-    show_info('Empty mask of Saturn ring radius below 10 kkm', test)
+#===============================================================================
+def exercise_backplanes_settings(obs):
+    """Wrapper for exercise_backplanes(). Determines default directories
+       based on input observation object and fills in args based on
+       settings."""
+
+    # determine default output and reference directories
+    specname = obs.filespec.split( \
```

Comment:
This still doesn't work if the TESTDATA_PARENT_DIRECTORY doesn't end with a /.

Reply:
The way I handle optional trailing slashes in directory paths is dir.rstrip('/') if the trailing slash should not be there and dir.rstrip('/') + "/" if it is required. Of course, os.path.join is the proper way to handle this, but sometimes I find that too cumbersome.

By the way, you never need a backslash after an opening paren.

Reply:
Ok, I see the same misbehavior as Rob using his definition of $OOPS_TEST_DATA_PATH, so I just used os.path.normpath() to standardize TESTDATA_PARENT_DIRECTORY where it is defined in oops/unitester_support, and included the trailing slash.  All other references to TESTDATA_PARENT_DIRECTORY are as an argument to os.path.join() so they don't care.